pagopa/io-web-profile
# [IOPID-602] : Adjustment for build static export

## Short description

The need arose to modify the project so that it could correctly generate the export of the project into static files (with internalization). In the current mode, it was being compiled as a Static Site Generator (SSG), project exported with (html, css, js files).

Note: With this approach, there will be no way to identify the default language, but there will be the path prefix /it/ or /lang/.

In comparison to what is described in the next-intl documentation

https://next-intl-docs.vercel.app/docs/routing/middleware#usage-without-middleware-static-export

This way, the export was resolved to obtain the desired static package with HTML and JavaScript files.
## List of changes proposed in this pull request

- Stored landing locale in local storage (necessary for redirect callback after spid login)
- Adjustment according to next-intl documentation for static next export
- Layout file modification according to next export docs
- Usage of next-intl components that wrap other components like LINK, useRouter to use relative paths with the locale
- Appropriate next-config
- Removal of middleware not supported in SSG export mode
- Other adjustments


## How to test

Generate the project in static mode using yarn build. Once the project is created in the .out folder, verify the presence of HTML files. Launch the build using yarn run start-static and verify the correct functioning of the portal.
